receiptui/src/main/java/com/hyperwallet/android/ui/receipt/view/TabbedListReceiptsFragment.java

```diff
@@ -0,0 +1,235 @@
+                getResources());
+        mListReceiptsViewPagerAdapter.addInitialItem(null);
+        mViewPager.setAdapter(mListReceiptsViewPagerAdapter);
+        mTabLayout.setupWithViewPager(mViewPager);
+        mTabbedListReceiptsViewModel.initialize();
+        registerObservers();
+    }
+
+    private void registerObservers() {
+        mTabbedListReceiptsViewModel.mPrepaidCards.observe(this, new Observer<List<PrepaidCard>>() {
+            @Override
+            public void onChanged(List<PrepaidCard> prepaidCards) {
+                if (!prepaidCards.isEmpty()) {
+                    boolean isWalletModel =
+                            mTabbedListReceiptsViewModel.mProgramModel.name() == ProgramModel.WALLET_MODEL.name();
+                    boolean isCardModel =
+                            mTabbedListReceiptsViewModel.mProgramModel.name() == ProgramModel.PAY2CARD_MODEL.name()
+                                    || mTabbedListReceiptsViewModel.mProgramModel.name()
+                                    == ProgramModel.CARD_ONLY_MODEL.name();
+
+                    if (isWalletModel) {
+                        mTabLayout.setVisibility(View.VISIBLE);
+                        sortToken(prepaidCards);
+                        for (PrepaidCard prepaidCard : prepaidCards) {
+                            mListReceiptsViewPagerAdapter.addItem(prepaidCard);
+                        }
+                    } else if (isCardModel) {
+                        if (prepaidCards.size() > 1) {
+                            mTabLayout.setVisibility(View.VISIBLE);
+                            sortToken(prepaidCards);
+                            for (PrepaidCard prepaidCard : prepaidCards) {
+                                mListReceiptsViewPagerAdapter.addItem(prepaidCard);
+                            }
+                        } else {
+                            mTabLayout.setVisibility(View.GONE);
+                            mListReceiptsViewPagerAdapter.addItem(prepaidCards.get(0));
+                        }
+                    }
+                    mListReceiptsViewPagerAdapter.notifyDataSetChanged();
+                } else {
+                    mTabLayout.setVisibility(View.GONE);
+                }
+            }
+        });
+
+    }
+
+    private void sortToken(List<PrepaidCard> prepaidCards) {
+        Collections.sort(prepaidCards, new Comparator<PrepaidCard>() {
+            @Override
+            public int compare(PrepaidCard firstPrepaid, PrepaidCard secondPrepaid) {
+                if (firstPrepaid == null || secondPrepaid == null
+                        || firstPrepaid.getPrimaryCardToken() == null || secondPrepaid.getPrimaryCardToken() == null) {
+                    return 0;
+                }
+                return firstPrepaid.getPrimaryCardToken().compareTo(secondPrepaid.getPrimaryCardToken());
+            }
+        });
+    }
```

Comment:
method name doesn't match its behavior, it should be `sortPrepaidCards`
Simplify the logic here, refer to how its done in App

Reply:
Can't skip the null checks and to simply this implementation Java8 is required which is not supported below Android 7.0. Should introduce comparator to new class?